### PR TITLE
[systemtest] Fix nightlies

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- Bugfix

### Description

This PR gonna fix failing tests from our nightly builds.

`AllNamespaceST&MultipleNamespaceST` - here was two problems - first one was same like in `TopicST` - there was problem in using `cmd` - already fixed by different PR, the second problem was in naming of topic - after refactor we waited for topic that had randomly generated name, but in our `examples` KafkaTopic yaml we have `my-topic` name -> so the randomly named topic wasn't never created

`ConnectS2IST` - here was problem in getting the right pod where KafkaConnector is receiving messages to sink -> the `connectorPod` had KafkaConnect build pod name, so the test failed because no messages was received to sink-file 

I also added deletion of all CR that are created without wait to prevent future problems

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=AllNamespaceST,MultipleNamespaceST,io.strimzi.systemtest.topic.TopicST,ConnectS2IST,io.strimzi.systemtest.metrics.MetricsST

File: systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java [hunk 1]
```diff
@@ -167,7 +167,7 @@ void testKafkaExporterDataAfterExchange() {
             assertThat("Value from collected metric should be non-empty", !value.isEmpty());
             assertThat("Metrics doesn't contain specific values", value.contains("kafka_consumergroup_current_offset"));
             assertThat("Metrics doesn't contain specific values", value.contains("kafka_consumergroup_lag"));
-            assertThat("Metrics doesn't contain specific values", value.contains("kafka_topic_partitions{topic=\"" + TEST_TOPIC_NAME + "\"} 7"));
+            assertThat("Metrics doesn't contain specific values", value.contains("kafka_topic_partitions{topic=\"" + TEST_TOPIC_NAME + "\"}"));
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java:170]: Why did you remove the number?

Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java:170]: The metrics with `"kafka_topic_partitions{topic=\"" + TEST_TOPIC_NAME + "\"}"` was there, but every time with different number at the end. Is that `7` needed there?

**Comment by im-konge** [comment]
These tests:
```
    testUserOperatorMetrics in io.strimzi.systemtest.metrics.MetricsST
    testTopicOperatorMetrics in io.strimzi.systemtest.metrics.MetricsST
    testDeployS2IWithMongoDBPlugin in io.strimzi.systemtest.ConnectS2IST
```
Are in investigating.

**Comment by im-konge** [comment]
Those tests will be fixed by @Frawless in next PR

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectS2IST#testKafkaConnectorWithConnectS2IAndConnectWithSameName,ConnectST#testKafkaConnectorWithConnectAndConnectS2IWithSameName,CustomResourceStatusST#testKafkaConnectorWithoutClusterConfig+testKafkaMirrorMaker2WrongBootstrap,io.strimzi.systemtest.kafka.ListenersST#testNonExistingCustomCertificate+testCertificateWithNonExistingDataCrt+testCertificateWithNonExistingDataKey,io.strimzi.systemtest.security.SecurityST#testTlsHostnameVerificationWithKafkaConnect+testTlsHostnameVerificationWithMirrorMaker+testKafkaAndKafkaConnectTlsVersion+testKafkaAndKafkaConnectCipherSuites,io.strimzi.systemtest.specific.SpecificST#testDeployUnsupportedKafka

**Comment by im-konge** [comment]
@strimzi-ci run tests profile=regression testcase=io.strimzi.systemtest.security.SecurityST#testTlsHostnameVerificationWithKafkaConnect,io.strimzi.systemtest.kafka.ListenersST#testCertificateWithNonExistingDataKey

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=io.strimzi.systemtest.security.SecurityST#testTlsHostnameVerificationWithKafkaConnect,io.strimzi.systemtest.kafka.ListenersST#testCertificateWithNonExistingDataKey

**Comment by im-konge** [comment]
@scholzj azure and travis should be green .. I'm trying that for x time. So it's up to you
